framework/core/js/src/common/extenders/Model.ts

```diff
@@ -0,0 +1,47 @@
+import IExtender, { IExtensionModule } from './IExtender';
+import Application from '../Application';
+import ActualModel from '../Model';
+
+export default class Model implements IExtender {
+  private model: { new (): ActualModel };
+  private type: string | null = null;
+
+  public constructor(model: { new (): ActualModel }) {
+    this.model = model;
+  }
+
+  public attribute<T, O = unknown>(name: string, transform: ((attr: O) => T) | null = null): Model {
```

Comment:
We might want to consider attaching things like this in the `extend` stage, so that we can perform safety checks / conflict resolution / "compilation analysis" of some sort.

Reply:
Agreed.